src/probnum/linops/_arithmetic.py

```diff
@@ -57,6 +94,295 @@ def matmul(op1: LinearOperator, op2: LinearOperator) -> LinearOperator:
 _mul_fns: _BinaryOperatorRegistryType = {}
 _matmul_fns: _BinaryOperatorRegistryType = {}
 
+########################################################################################
+# Fill Arithmetics Registries
+########################################################################################
+
+# Scaling
+def _mul_scalar_scaling(scalar: ScalarArgType, scaling: Scaling) -> Scaling:
+    if scaling.is_isotropic:
+        return Scaling(scalar * scaling.scalar, shape=scaling.shape)
+
+    return Scaling(scalar * scaling.factors, shape=scaling.shape)
+
+
+def _mul_scaling_scalar(scaling: Scaling, scalar: ScalarArgType) -> Scaling:
+    if scaling.is_isotropic:
+        return Scaling(scalar * scaling.scalar, shape=scaling.shape)
+
+    return Scaling(scalar * scaling.factors, shape=scaling.shape)
+
+
+_mul_fns[("scalar", Scaling)] = _mul_scalar_scaling
+_mul_fns[(Scaling, "scalar")] = _mul_scaling_scalar
+_add_fns[(Scaling, Scaling)] = Scaling._add_scaling
+_sub_fns[(Scaling, Scaling)] = Scaling._sub_scaling
+_mul_fns[(Scaling, Scaling)] = Scaling._mul_scaling
+_matmul_fns[(Scaling, Scaling)] = Scaling._matmul_scaling
+
+# ScaledLinearOperator
+def _matmul_scaled_op(scaled, anylinop):
+    return scaled._scalar * (scaled._linop @ anylinop)
+
+
+def _matmul_op_scaled(anylinop, scaled):
+    return scaled._scalar * (anylinop @ scaled._linop)
+
+
+for op_type in _AnyLinOp:
+    _matmul_fns[(ScaledLinearOperator, op_type)] = _matmul_scaled_op
+    _matmul_fns[(op_type, ScaledLinearOperator)] = _matmul_op_scaled
+    _matmul_fns[(NegatedLinearOperator, op_type)] = _matmul_scaled_op
+    _matmul_fns[(op_type, NegatedLinearOperator)] = _matmul_op_scaled
+
+
+# Kronecker
+
+
+def _matmul_scaling_kronecker(scaling: Scaling, kronecker: Kronecker) -> Kronecker:
+    if scaling.shape[1] != kronecker.shape[0]:
+        raise ValueError(
+            f"matmul received invalid shapes {scaling.shape} @ {kronecker.shape}"
+        )
+
+    if scaling.is_isotropic:
+        return scaling.scalar * kronecker
+    return NotImplemented
+
+
+def _matmul_kronecker_scaling(kronecker: Kronecker, scaling: Scaling) -> Kronecker:
+    if kronecker.shape[1] != scaling.shape[0]:
+        raise ValueError(
+            f"matmul received invalid shapes {kronecker.shape} @ {scaling.shape}"
+        )
+
+    if scaling.is_isotropic:
+        return kronecker * scaling.scalar
+    return NotImplemented
+
+
+def _mul_scalar_kronecker(scalar: ScalarArgType, kronecker: Kronecker) -> Kronecker:
+    sqrt_scalar = np.sqrt(scalar)
```

Comment:
This does not work for negative scalars, we will need some sort of workaround here :/